lib/Runtime/Library/JavascriptArray.h

```diff
@@ -1137,6 +1137,8 @@ namespace Js
         uint32 GetNextIndex(uint32 index) const;
         BOOL DirectGetItemAt(uint32 index, int* outVal);
 
+        virtual PropertyQueryFlags GetPropertyQuery(Var originalInstance, PropertyId propertyId, Var* value, PropertyValueInfo* info, ScriptContext* requestContext) override;
```

Comment:
Do we need one of these for HasPropertyQuery too?

Reply:
I don't think so, but maybe need one fore GetItemQuery (the other path that GetElementI can go to).

Reply:
Actually nvm, NativeIntArray already has a GetItemQuery which does the check. Also upon consideration, I'm going to remove these overloads and just add checks to `JavascriptArray::GetPropertyQuery`. It's not a hot path and seems better not to be changing the vtable like this.